ledger-core/v2/virtual/handlers/vdelegatedcallrequest.go

```diff
@@ -0,0 +1,214 @@
+		semaphoreReadyToWork smachine.SyncLink
+	)
+
+	action := func(state *object.SharedState) {
+		semaphoreReadyToWork = state.ReadyToWork
+	}
+
+	switch s.objectSharedState.Prepare(action).TryUse(ctx).GetDecision() {
+	case smachine.Passed:
+	case smachine.NotPassed:
+		return ctx.WaitShared(s.objectSharedState.SharedDataLink).ThenRepeat()
+	default:
+		panic(throw.Impossible())
+	}
+
+	if ctx.AcquireForThisStep(semaphoreReadyToWork).IsNotPassed() {
+		return ctx.Sleep().ThenRepeat()
+	}
+
+	return ctx.Jump(s.processRequest)
+}
+
+func (s *SMVDelegatedCallRequest) processRequest(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	var (
+		tooOldRequest bool
+
+		pendingTableFull bool
+	)
+
+	action := func(state *object.SharedState) {
+		var (
+			oldestPulse  pulse.Number
+			pendingList  *object.RequestList
+			pendingCount uint8
+		)
+
+		callTolerance := s.Payload.CallFlags.GetInterference()
+
+		switch callTolerance {
+		case contract.CallTolerable:
+			pendingList = state.PendingTable.GetList(contract.CallTolerable)
+			oldestPulse = state.OrderedPendingEarliestPulse
+			pendingCount = state.ActiveOrderedPendingCount
+		case contract.CallIntolerable:
+			pendingList = state.PendingTable.GetList(contract.CallIntolerable)
+			oldestPulse = state.UnorderedPendingEarliestPulse
+			pendingCount = state.ActiveUnorderedPendingCount
+		default:
+			panic(throw.Unsupported())
+		}
+
+		if oldestPulse == pulse.Unknown || s.Payload.RequestReference.GetLocal().GetPulseNumber() < oldestPulse {
+			tooOldRequest = true
+			return
+		}
+
+		// pendingList already full
+		if pendingList.Count() == int(pendingCount) && !pendingList.Exist(s.Payload.RequestReference) {
+			pendingTableFull = true
+			return
+		}
+
+		if !pendingList.Add(s.Payload.RequestReference) {
+			ctx.Log().Trace(struct {
+				*log.Msg  `txt:"request already in pending list"`
+				Reference string
+			}{Reference: s.Payload.RequestReference.String()})
+			// request was already in the list we will delegate token, maybe it is repeated call
+			return
+		}
+
+		if pendingList.Count() == int(pendingCount) {
+			state.SetPendingListFilled(ctx, callTolerance)
+		}
+	}
+
+	switch s.objectSharedState.Prepare(action).TryUse(ctx).GetDecision() {
+	case smachine.Passed:
+	case smachine.NotPassed:
+		return ctx.WaitShared(s.objectSharedState.SharedDataLink).ThenRepeat()
+	default:
+		panic(throw.Impossible())
+	}
+
+	if tooOldRequest {
+		ctx.Log().Warn(struct {
+			*log.Msg  `txt:"There is no such object"`
+			Reference string
+		}{Reference: s.Payload.RequestReference.String()})
+		return ctx.Error(throw.IllegalValue())
+	}
+
+	if pendingTableFull {
+		ctx.Log().Warn(struct {
+			*log.Msg  `txt:"Pending table already full"`
+			Reference string
+		}{Reference: s.Payload.RequestReference.String()})
+		return ctx.Error(throw.IllegalState())
+	}
+
+	return ctx.Jump(s.buildResponse)
+}
+
+func (s *SMVDelegatedCallRequest) buildResponse(ctx smachine.ExecutionContext) smachine.StateUpdate {
```

Comment:
please, use step prefix for steps: `s.stepGetDelegation`, `s.stepBuildResponse`

Reply:
done